source_modelling/sources.py

```diff
@@ -339,31 +356,125 @@ def from_centroid_strike_dip(
         Parameters
         ----------
         centroid : np.ndarray
-            The centre of the fault plane in lat, lon coordinates.
-        strike : float
-            The strike of the fault (in degrees).
-        dip_dir : Optional[float]
-            The dip direction of the fault (in degrees). If None this is assumed to be strike + 90 degrees.
-        top : float
-            The top depth of the plane (in km).
-        bottom : float
-            The bottom depth of the plane (in km).
+            The centre of the fault plane in lat, lon, and optionally depth (km) coordinates.
+        dip : float
```

Comment:
Maybe just for extra clarity could add WGS84 bearing for dip and strike

Reply:
dip is not a bearing from north so it is not a WGS84 bearing. Strike and dip direction sure.

Reply:
Sorry meant dip_dir not dip